server/public/model/metrics.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"fmt"
+	"time"
+)
+
+type MetricType string
+
+const (
+	ClientMetricChannelVisited MetricType = "channel_visited"
+	ClientMetricChannelLoad    MetricType = "channel_load"
+
+	performanceReportTTLMilliseconds = 300 * 1000 // 300 seconds/5 minutes
+)
+
+type MetricSample struct {
+	Metric    MetricType        `json:"metric"`
+	Value     int64             `json:"value"`
+	Timestamp int64             `json:"timestamp,omitempty"`
+	Labels    map[string]string `json:"labels,omitempty"`
+}
+
+// PerformanceReport is a set of samples collected from a client
+type PerformanceReport struct {
+	Version    string            `json:"version"`
+	ClientID   string            `json:"client_id"`
+	Labels     map[string]string `json:"labels"`
```

Comment:
Why have this be a map instead of two separate fields for user agent and OS like before?

Reply:
I think we can define more labels rather than creating a newer version of the performance report. Adding labels should change the report version.

Reply:
I'm fine with that as long as we still validate the labels on the serverside to prevent a malicious client from creating extra labels